mkdocs_drawio_file/plugin.py

```diff
@@ -1,68 +1,79 @@
-# A global is required because plugins are reinitialized each time a change is detected.
-last_run_timestamp = 0
-
-    
-
-    
-class drawio_file_plugin(mkdocs.plugins.BasePlugin):
+import logging
+import mkdocs
+from bs4 import BeautifulSoup
+from mkdocs.plugins import BasePlugin
+
+
+# ------------------------
+# Constants and utilities
+# ------------------------
+RE_PATTERN = r'!\[(.*?)\]\((.*?.drawio)\)'
+SUB_TEMPLATE = string.Template(
+        "<div class=\"mxgraph\" style=\"max-width:100%;border:1px solid transparent;\" data-mxgraph=\"{&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;nav&quot;:true,&quot;resize&quot;:true,&quot;toolbar&quot;:&quot;zoom layers tags lightbox&quot;,&quot;edit&quot;:&quot;_blank&quot;,&quot;xml&quot;:&quot;$xml_drawio&quot;}\"></div>")
+
+# ------------------------
+# Plugin
+# ------------------------
+class DrawioFilePlugin(BasePlugin):
+    """
+    Plugin for embedding DrawIO Diagrams into your Docs
+    """
     config_scheme = (
         (
             "file_extension",
             mkdocs.config.config_options.Type(str, default=".drawio"),
         ),
     )
 
-    TEMPLATE = string.Template("<div class=\"mxgraph\" style=\"max-width:100%;border:1px solid transparent;\" data-mxgraph=\"{&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;nav&quot;:true,&quot;resize&quot;:true,&quot;toolbar&quot;:&quot;zoom layers tags lightbox&quot;,&quot;edit&quot;:&quot;_blank&quot;,&quot;xml&quot;:&quot;$xml_drawio&quot;}\"></div>")
-
     def __init__(self):
         self.log = logging.getLogger("mkdocs.plugins.diagrams")
         self.pool = None
 
-
-
-    def escape( str_xml: str ):
+    def escape(self, str_xml: str):
         str_xml = str_xml.replace("&", "&amp;")
         str_xml = str_xml.replace("<", "&lt;")
         str_xml = str_xml.replace(">", "&gt;")
         str_xml = str_xml.replace("\"", "\&quot;")
         str_xml = str_xml.replace("'", "&apos;")
         return str_xml
 
-    def conver_file( file_name: str):
+    def substitute_image(self, path, src: str):
+        if src.startswith("../"):
+            src = src[3:]
+
+        file_name = os.path.join(path, src)
+
         with open(file_name, 'r') as q_data:
             q_lines = q_data.readlines()
 
-        drawio_text = ''.join([ item.strip() for item in q_lines])
+        drawio_text = ''.join([item.strip() for item in q_lines])
+        drawio_text_ecaped = self.escape(drawio_text)
+
+        return SUB_TEMPLATE.substitute(xml_drawio=drawio_text_ecaped)
+
+    def on_post_page(self, output_content, config, page, **kwargs):
+        if ".drawio" not in output_content.lower():
+            # Skip unecessary HTML parsing
+            return output_content
+
+        soup = BeautifulSoup(output_content, 'html.parser')
 
-        drawio_text_ecaped = drawio_file_plugin.escape(drawio_text)
+        # search for images using drawio extension
+        diagrams = soup.findAll('img', src=re.compile('.*\.drawio', re.IGNORECASE))
```

Comment:
I believe that this changed line has introduced the issue #7 

Because the trailing `)`, which is now in `RE_PATTERN` and was also present in the line 64 of the old version,
is gone now.
And therefore the string `[test](test.drawio.png)` matches the pattern.